File: Back/Apis/Api_Py/src/models/tablas_sql/tabla_llaves_privadas.py
#Librerias Instaladas
from sqlalchemy import ForeignKey, String;
from sqlalchemy.orm import Mapped, mapped_column, relationship;

#Tablas Creadas
from src.models.tablas_sql.tabla_base import Tabla_Base;

#Modulos Nativos
from typing import List, Optional;
from datetime import datetime;

class Tabla_Llaves_Privadas(Tabla_Base):
    __tablename__ = 'llaves_privadas';

    llave_id: Mapped[int] = mapped_column(nullable=False, primary_key=True, autoincrement=True);
    llave_valor: Mapped[str] = mapped_column(String(1000),nullable=False);
    firma_id_fk: Mapped[int] = mapped_column(ForeignKey('firmas.firma_id'), nullable=False);

    restriccion_firmas_en_llaves: Mapped['Tabla_Firmas'] = relationship(back_populates='restriccion_firmas_para_llaves');

    def __repr__(self) -> str:
        return f"{self.llave_id!r},{self.llave_valor!r},{self.firma_id_fk!r}"

File: Back/Apis/Api_Js/src/modules/baseDatos/tablas/tabla_llaves_privadas.js
//Librerias
import { DataTypes } from "sequelize";

//Configuraciones Creadas
import { conexionBaseDatos } from "../../../settings/consultas_sql/conexion_sql.js";

export const tabla_llaves_privadas = conexionBaseDatos.define(
    "llaves_privadas",
    {
        llave_id:{
            type: DataTypes.INTEGER,
            primaryKey: true,
            allowNull: false,
            autoIncrement: true,
        }, 
        llave_valor:{
            type:DataTypes.STRING,
            allowNull:false,
            unique:true
        },
        firma_id_fk:{
            type:DataTypes.INTEGER,
            allowNull:false,
            unique:true
        }
    },
    {
        freezeTableName: true,
        timestamps: false,
    }
)